src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -307,6 +316,89 @@ class GroupQueryAttentionDecomposition : public ov::pass::MatcherPass {
     }
 };
 
+class RedirectNewKvToOutput : public ov::pass::MatcherPass {
+public:
+    // context len of second concat operator - should be 1 or equal first operand
+    RedirectNewKvToOutput() {
+        auto match_down_up_convert_subgraph = [](const ov::Output<ov::Node>& input) {
+            auto upconvert =
+                ov::pass::pattern::wrap_type<ov::op::v0::Convert>({input},
+                                                                  ov::pass::pattern::type_matches(ov::element::f32));
+
+            auto upscale = ov::pass::pattern::wrap_type<ov::op::v0::Constant>(ov::pass::pattern::rank_equals(0));
+            auto upmul = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({upconvert, upscale});
+
+            auto downscale = ov::pass::pattern::wrap_type<ov::op::v0::Constant>(ov::pass::pattern::rank_equals(0));
+            auto downmul = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({upmul, downscale});
+
+            auto downconvert = ov::pass::pattern::wrap_type<ov::op::v0::Convert>(
+                {downmul},
+                ov::pass::pattern::type_matches_any({ov::element::f8e4m3, ov::element::f8e5m2}));
+            // TODO: need to check that upscale*downscale = 1
+            // TODO: need to check input type is f8e5m2 or f8e4m3 if we use this version of concat
+
+            return downconvert;
+        };
+
+        // input0 : float8e4m3[1,32,1151,96]
+        // input1 : float8e4m3[1,32,1,96]
+        // TODO: might be transposed - so no assumptions on number of tokens can be made:
+        auto input0 =
+            ov::pass::pattern::any_input(ov::pass::pattern::shape_matches("[1, ?, CTX_LEN,  CTX_TRANSPOSED_LEN]"));
+        auto input1 =
+            ov::pass::pattern::any_input(ov::pass::pattern::shape_matches("[1, ?, CTX2_LEN, CTX2_TRANSPOSED_LEN]"));
```

Comment:
I like these code a lot, however we can name them `past` and `present` kv input safely too. But is a minor comment